torch/lib/THC/generic/THCTensor.c

```diff
@@ -334,8 +334,8 @@ void THCTensor_(expandNd)(THCState *state, THCTensor **rets, THCTensor **ops, in
     THArgCheck(THCTensor_(nDimension)(state, ops[i]) > 0, i, "can't expand empty tensor %d", i);
   }
 
-  long *op_sizes[count];
-  long op_dims[count];
+  int64_t **op_sizes = (int64_t **)malloc(count * sizeof(int64_t *));
```

Comment:
If you do a `malloc` you need to free it afterwards.
We have functions for the `malloc`/`free` that you could use (`THAlloc` and `THFree`).